awslabs/aws-iot-device-client
# Updated SDK commit to use the latest SDK code

### Motivation
- Please give a brief description for the background of this change.
- Issue number: #411 


### Modifications
#### Change summary
* Updated SDK commit to use the latest SDK code.
* Updated resource manager `initializeAllocator` method to public so it can be called from outside of `SharedCrtResourceManager`. 
* Added null check in `SharedCrtResourceManager::disconnect` method. 
* Updated unit test code to initialize allocator before using CRT library methods like CJSON or Crt::Map.
* Reverted this change to avoid calling `init` twice when FP is enabled. https://github.com/awslabs/aws-iot-device-client/pull/391/files

#### Revision diff summary
If there is more than one revision, please explain what has been changed since the last revision.
* Addressed reviewers comments 
* Added null check in `SharedCrtResourceManager::disconnect` method. 
* Updated unit test code to initialize allocator before using CRT library methods like CJSON or Crt::Map.

### Testing
 **Is your change tested? If not, please justify the reason.**  
 **Please list your testing steps and test results.** 
- Manually tested changes locally 
- Ran CI tests 
- Ran Unit tests locally. 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
